docs/design/rubygems.md

```diff
@@ -0,0 +1,522 @@
+of the repository placed in a folder. For Git dependencies, the folder name must match this specific
+[format](https://github.com/rubygems/rubygems/blob/3da9b1dda0824d1d770780352bb1d3f287cb2df5/bundler/lib/bundler/source/git.rb#L130):
+
+```ruby
+"#{base_name}-#{shortref_for_path(revision)}"
+```
+
+The name of the directory **must come from the Git URL**, not the actual name of the gem, and the cloned folder must
+contain unpacked source code. Any other format will cause bundler to try to re-download the repository, causing the
+build to fail.
+
+##### Multiple Gems in a single repository
+
+A single repository can hold multiple Gems, and those can be imported as dependencies. When this happens, Bundler still
+expects a single clone to be made. Here's an example of how multiple gems imported from a single repository+revision
+looks like in the `Gemfile.lock`:
+
+```
+GIT
+  remote: https://github.com/chatwoot/azure-storage-ruby
+  revision: 9957cf899d33a285b5dfe15bdb875292398e392b
+  branch: chatwoot
+  specs:
+    azure-storage-blob (2.0.3)
+      azure-storage-common (~> 2.0)
+      nokogiri (~> 1, >= 1.10.8)
+    azure-storage-common (2.0.4)
+      faraday (~> 2.0)
+      faraday-follow_redirects (~> 0.3.0)
+      faraday-net_http_persistent (~> 2.0)
+      net-http-persistent (~> 4.0)
+      nokogiri (~> 1, >= 1.10.8)
+```
+
+### Out of scope
+
+#### Plugins
+Bundler has support for using [plugins](https://bundler.io/guides/bundler_plugins.html), which allows users to extend
+Bundler's functionality in any way that they seem fit. Since this can open the possibility for security issues, plugins
+will not be supported by Cachi2.
+
+Since we're not proposing the direct usage of Bundler to fetch the dependencies, no other actions are needed in the
+prefetch phase, existing plugin definitions will be ignored.
+
+#### Pre-compiled binaries
+For the initial implementation, we're aiming to provide support only for plain Ruby gems (which are idenfied as `RUBY`
+in the `PLATFORMS` section of the `Gemfile.lock`). This means that we need to explicitly forbid the processing of any
+repository that has references to any platform besides `RUBY`.
+
+To achieve this, we can implement a check in Cachi2 that will cause the request to fail if any other platform is
+specified in the `Gemfile.lock`. This way, we can avoid downloading any pre-compiled content.
+
+We should probably implement support for pre-compiled binaries as a follow-up feature, similarly to what was done with
+[pip wheels](https://github.com/containerbuildsystem/cachi2/blob/main/docs/pip.md#distribution-formats).
+
+#### Checksum verification
+Since checksums in the `Gemlock.file` is still a feature in development (see [checksums](#dependency-checksums)), we
+can postpone implementing support for it until the feature is delivered.
+
+We need to decide if we will report all dependencies as having missing checksums in the SBOM, or not.
```

Comment:
I would say yes, but only if `BUNDLED WITH >= $the_version_that_will_officially_add_support`

Reply:
We can probably file a story to keep an eye on this feature, and update our implementation when it is released.

Reply:
I feel like noting missing checksums should be an **SBOM** feature which is independent of specific package managers.

If nothing else, we're going to have a situation at some point in the future where existing (old) ruby SBOMS **without** checksums look exactly like current ruby SBOMs **with** checksums (because SBOMs simply ignore checksums if they exist)...

Reply:
P.S. Trust me that it **can be** quite difficult to implement tracking checksum presence and responsible file(s) *after the initial implementation* (ask me how I know :grimacing:)

Reply:
So you're saying we should mark all Ruby components as `missing_checksum_in_file`?

Reply:
Yes

Reply:
Remember this is ultimately a user's decision, so IIUC this whole thing, unless users opt-in explicitly or use a new enough bundler, then in order to **always** report checksums, we'd not only have to opt-in that behind users's backs we'd also dirty their repo by modifying the lockfile, am getting it right? If so, then I'm also against such practice and instead report that in the SBOM, after all, this should be user's responsibility to better harden what their source code depends on, we're only a messenger here really.

Reply:
No.

> in order to always report checksums

We'll **only** report that they are *missing*, which requires nothing from the user. If the user *starts* using checksums, we'll no longer add the "missing" annotation to the SBOMs. That's all.

Reply:
It would be to rude report that checksums are missing if the package manager doesn't even allow the user to specify checksums. And bundler currently doesn't (not officially, and I don't think our documentation should recommend that users do something their tool treats as tech preview)

Reply:
OK :shrug:.

From my POV, it's more important to have consistency across SBOMs regardless of package manager, particularly given that we **only** report checksums if they are *missing* (it would be completely different if we reported presence AND absence).

Now an SBOM which doesn't mention checksums will be "either *there are checksums* (which we don't report), or *there aren't checksums, but we're not reporting it*", which is so inconsistent as to be meaningless.

I'm happy to give way to the majority here, though.

Reply:
Resolving this since a conclusion was reached.